packages/amplify-graphql-model-transformer/src/graphql-model-transformer.ts

```diff
@@ -213,109 +231,24 @@ export class ModelTransformer extends TransformerModelBase implements Transforme
 
       // Update the field with auto generatable Fields
       this.addAutoGeneratableFields(ctx, type);
+
+      if (this.options.SyncConfig) {
+        this.addModelSyncFields(ctx, type);
+      }
```

Comment:
we should use the datastore check here if we can (note: the interfaces will need this function as well)
```suggestion
      if (ctx.isProjectUsingDataStore()) {
        this.options.SyncConfig = SyncUtils.getSyncConfig(ctx as TransformerContextProvider, def!.name.value);
        this.addModelSyncFields(ctx, type);
      }
```